src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -2,6 +2,54 @@
 
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int[] arrayFrom = getIntArray(dateFrom.split("\\D"));
+        int[] arrayTo = getIntArray(dateTo.split("\\D"));
+        StringBuilder sb = new StringBuilder("Report for period ");
+        sb.append(dateFrom).append(" - ").append(dateTo).append(System.lineSeparator());
+
+        for (int i = 0; i < names.length; i++) {
+            sb.append(names[i]).append(" - ");
+            int sumName = 0;
+            for (int j = 0; j < data.length; j++) {
+
+                String[] words = data[j].split(names[i]);
+                int[] actualDate = getIntArray(words[0].split("\\D"));
+
+                if (words.length > 1) {
+                    int[] salary = getIntArray(words[1].split("\\D"));
+                    if (arrayTo[1] == arrayFrom[1]) {
+                        if (actualDate[0] >= arrayFrom[0] && actualDate[0] <= arrayTo[0]) {
+                            sumName = replaceSalary(sumName, salary);
+                        }
+                    } else {
+                        if (actualDate[1] == arrayFrom[1] && actualDate[0] >= arrayFrom[0]) {
+                            sumName = replaceSalary(sumName, salary);
+                        } else if (actualDate[1] == arrayTo[1] && actualDate[0] <= arrayTo[0]) {
+                            sumName = replaceSalary(sumName, salary);
+                        }
+                    }
```

Comment:
this logic looks too complicated. We can just get array with data from row, get date data by index and transform date with formatted to LocalDate and compare dates and names.